federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java

```diff
@@ -383,11 +383,19 @@ public void onCreate(KeycloakSession session, RealmModel realm, ComponentModel m
         } else {
             mapperModel = KeycloakModelUtils.createComponentModel("first name", model.getId(), UserAttributeLDAPStorageMapperFactory.PROVIDER_ID,LDAPStorageMapper.class.getName(),
                     UserAttributeLDAPStorageMapper.USER_MODEL_ATTRIBUTE, UserModel.FIRST_NAME,
-                    UserAttributeLDAPStorageMapper.LDAP_ATTRIBUTE, LDAPConstants.CN,
+                    UserAttributeLDAPStorageMapper.LDAP_ATTRIBUTE, LDAPConstants.GIVENNAME,
                     UserAttributeLDAPStorageMapper.READ_ONLY, readOnly,
                     UserAttributeLDAPStorageMapper.ALWAYS_READ_VALUE_FROM_LDAP, alwaysReadValueFromLDAP,
                     UserAttributeLDAPStorageMapper.IS_MANDATORY_IN_LDAP, "true");
             realm.addComponentModel(mapperModel);
+
+            if (editMode == UserStorageProvider.EditMode.WRITABLE) {
```

Comment:
If `Edit Mode` is `READ_ONLY` we still want to map this attribute from LDAP, don't we? Or is there a reason to have it only if writable?